mypy/checkexpr.py

```
@@ -1490,6 +1490,27 @@ def check_for_extra_actual_arguments(self,
                                                                         context)
                             is_unexpected_arg_error = True
                         ok = False
+                elif (isinstance(actual_type, Instance) and
+                        actual_type.type.has_base('typing.Mapping')):
+                    any_type = AnyType(TypeOfAny.special_form)
+                    mapping_info = self.chk.named_generic_type('typing.Mapping',
+                                                               [any_type, any_type]).type
+                    supertype = map_instance_to_supertype(actual_type, mapping_info)
+                    if messages and supertype.args:
```

Ruff (0.8.2)

Undefined name `messages`

(F821)